xbmc/guilib/guiinfo/MusicGUIInfo.cpp

```diff
@@ -503,11 +511,21 @@ bool CMusicGUIInfo::GetInt(int& value, const CGUIListItem *gitem, int contextWin
 
 bool CMusicGUIInfo::GetBool(bool& value, const CGUIListItem *gitem, int contextWindow, const CGUIInfo &info) const
 {
+  const CFileItem *item = static_cast<const CFileItem*>(gitem);
+  const CMusicInfoTag* tag = item->GetMusicInfoTag();
+
   switch (info.m_info)
   {
     ///////////////////////////////////////////////////////////////////////////////////////////////
     // MUSICPLAYER_*
     ///////////////////////////////////////////////////////////////////////////////////////////////
+    case LISTITEM_IS_BOXSET:
```

Comment:
Treating bool correctly - great point, I didn't spot that at all. :)
Sorry to nit-pick, but could you move it under a // LISTITEM_* section heading comment

Also wondering if The CFileitem and CMusicInfoTag only get created for that too rather than for every music bool, it could then be moved if other LISTITEM_ bools ever get added. Not sure on the impact of creating those things each time.

Reply:
Yep, moved it under it's own heading comment.
It follows the same methodology as the video database, but if you're worried about impact, how about this ? 
`case LISTITEM_ISBOXSET: `
`const CFileItem *item = static_cast<const CFileItem*>(gitem);`
`      const CMusicInfoTag* tag = item->GetMusicInfoTag();`
`      if (tag)                                                                               `
`        {                                                                                             `
`        value = item->GetMusicInfoTag()->GetBoxset() == true;`
`        return true;`
`      }`
`      break;`
So only create the item and get the tag if we want that particular bool, rather than create it for any bool.  Less overhead if a skin is only checking the other player related bools.

Reply:
If tag position is same as video then fine be consistent with that

Reply:
The position of the creation of the item and tag was consistent with video not necessarily the position of the listitem under a heading.  It's still going to be less overhead (even if only slightly), creating the item and fetching the tag only when needed.

Reply:
Happy either way, but yes your code snippet is what I had in mind in my comment